pcsx2/gui/AppUserMode.cpp

```diff
@@ -15,296 +15,332 @@
 
 #include "PrecompiledHeader.h"
 #include "MainFrame.h"
-#include "Utilities/IniInterface.h"
+#include "Utilities/PathUtils.h"
 #include "Dialogs/ModalPopups.h"
-
+#include "yaml-cpp/yaml.h"
+#include "AppConfig.h"
 #include <wx/stdpaths.h>
 
+
 #ifdef __WXMSW__
 #include "wx/msw/regconf.h"
 #endif
 
 DocsModeType			DocsFolderMode = DocsFolder_User;
 bool					UseDefaultSettingsFolder = true;
 bool					UseDefaultPluginsFolder = true;
+bool 					runWizard  = true; // This should default to true unless the stream says otherwise. If this doesn't get flipped somethings wrong and setup required
+
+std::vector<std::string> ErrorFolders;
+
+std::string data;
+std::string usermodePath;
 
+fs::path CustomDocumentsFolder;
+fs::path SettingsFolder;
 
-wxDirName				CustomDocumentsFolder;
-wxDirName				SettingsFolder;
+std::string InstallFolder;
+fs::path PluginsFolder;
 
-wxDirName				InstallFolder;
-wxDirName				PluginsFolder;
+YAML::Node stream;
+FolderUtils folderUtils;
+
+
+const std::string PermissionFolders[] =
+	{
+		"settings",
+		"memcards",
+		"sstates",
+		"snapshots",
+		"logs",
+		"cheats_ws",
+#ifdef PCSX2_DEVBUILD
+		"dumps",
+#endif
+};
```

Comment:
Why drop PathDefs?

Reply:
PathDefs just seemed like extra bloat for something that would be an easy thing to fit in AppUserMode where that's actually used.